tests/bluetooth/audio/ccid/src/main.c

```diff
@@ -0,0 +1,87 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/bluetooth/audio/ccid.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/fff.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/util_macro.h>
+
+#include <ztest_test.h>
+#include <ztest_assert.h>
+
+DEFINE_FFF_GLOBALS;
+
+ZTEST_SUITE(audio_ccid_test_suite, NULL, NULL, NULL, NULL, NULL);
+
+#define MAX_CCID_CNT 256
+
+ZTEST(audio_ccid_test_suite, test_bt_ccid_alloc_value)
+{
+	/* Verify that we can properly wrap CCID allocation */
```

Comment:
I don't actually want to add anything new, I just prefer that it is also documented in the function @brief.

Since we testing this functionality with unit tests, it seems to me like a commited part of how the funciton is supposed to work. I think it should be documented in function signature, so it can be seen immediatly, and not only inside the actual function scope, where the developer might not look immediatly.

Reply:
I'm still not sure I follow :D 

The function provides a CCID. That is documented. Whether it does that by a loop or takes a random value is not really relevant for the application, and documenting it would prevent us from easily modifying the function. 

This test here does test the current looping behavior to ensure that we can allocate more than the max number of CCIDs if they are not assigned. That is also documented in the function:
> Allocating a CCID without registering the characteristic may (in very rare cases) result in duplicated CCIDs on the device.

Reply:
Okay, I see that I'm explaining my concern very poorly now.

So the function signature claims:
> * This should always be called right before registering a GATT service that contains a
> * @ref BT_UUID_CCID characteristic. Allocating a CCID without registering the characteristic
> * may (in very rare cases) result in duplicated CCIDs on the device.

In reality, what could happen if you do not follow this and skip registering the service, is that your allocation is lost.
For example:

-Assume that max CCID count is set to two.
-User then registers three 2 CCIDs (and we have used all allocations)
-User then registers a third one, this will overwrite the first one

For the final step, it is not documented this will happen. The user could assume that rather than overwriting a value an error code was expected. I.e. the allocation is only final once the service has been registered. Right?
Rather the signature hints the above will not happen, as it says `may (in very rare cases) result in duplicated CCIDs on the device.`

Reply:
I see, I misunderstood what happens here then.

But then, what is the purpose of this tests, and why is the behavior legal?